front/src/applications/operationalStudies/utils.ts

```diff
@@ -0,0 +1,320 @@
+        ) as ElectricalRangesData<ElectricalProfileValue>;
+
+        acc.push({
+          electrificationUsage: {
+            ...currentElectrification.values,
+            ...associatedProfile.values,
+          },
+          start: currentElectrification.start,
+          stop: currentElectrification.stop,
+        });
+      }
+      // If we reached the end of the electricalProfiles array, we use its last value for the profile
+    } else {
+      // Find the profile matching the current electrification range
+      // We know we will find one because theirs last stops are the same
+      const associatedProfile = electricalProfiles.find(
+        (profile) => profile.stop >= currentElectrification.stop
+      ) as ElectricalRangesData<ElectricalProfileValue>;
+
+      acc.push({
+        electrificationUsage: {
+          ...currentElectrification.values,
+          ...associatedProfile.values,
+        },
+        start: currentElectrification.start,
+        stop: currentElectrification.stop,
+      });
+    }
+
+    return acc;
+  }, []);
+
+/**
+ * Format path propreties data to be used in simulation results charts
+ */
+export const preparePathPropertiesData = (
+  electricalProfiles: SimulationResponseSuccess['electrical_profiles'],
+  { slopes, curves, electrifications, operational_points, geometry }: PathProperties,
+  pathLength: number
+): PathPropertiesFormatted => {
+  const formattedSlopes = transformBoundariesDataToPositionDataArray(
+    slopes as NonNullable<PathProperties['slopes']>,
+    pathLength,
+    'gradient'
+  );
+
+  const formattedCurves = transformBoundariesDataToPositionDataArray(
+    curves as NonNullable<PathProperties['curves']>,
+    pathLength,
+    'radius'
+  );
+
+  const electrificationsRanges = transformBoundariesDataToRangesData(
+    electrifications as NonNullable<PathProperties['electrifications']>,
+    pathLength
+  ) as ElectricalRangesData<ElectrificationValue>[];
+
+  const electricalProfilesRanges = transformBoundariesDataToRangesData(
+    electricalProfiles,
+    pathLength
+  ) as ElectricalRangesData<ElectricalProfileValue>[];
+
+  const electrificationRanges = formatElectrificationRanges(
+    electrificationsRanges,
+    electricalProfilesRanges
+  );
+
+  return {
+    electrifications: electrificationRanges,
+    curves: formattedCurves,
+    slopes: formattedSlopes,
+    operationalPoints: operational_points as NonNullable<PathProperties['operational_points']>,
+    geometry: geometry as NonNullable<PathProperties['geometry']>,
+  };
+};
+
+/**
+ * Return the power restriction codes of the rolling stock by mode
+ *
+ * ex: { "1500V": ["C1US", "C2US"], "25000V": ["M1US"], "thermal": []}
+ */
+export const getRollingStockPowerRestrictionsByMode = (
+  rollingStockModes: RollingStock['effort_curves']['modes']
+): { [mode: string]: string[] } => {
+  const curvesModesKey = Object.keys(rollingStockModes);
+
+  return reduce(
```

Comment:
Use the array's method directly: 
```suggestion
  return curvesModesKey.reduce(
```

Reply:
Done